sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultSetting.java

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.administration.models;
+
+import com.azure.core.annotation.Immutable;
+
+/**
+ * The {@link KeyVaultSetting} model.
+ */
+@Immutable
+public final class KeyVaultSetting {
+    private String name;
+    private String value;
+    private KeyVaultSettingType type;
+
+    public KeyVaultSetting(String name, String value, KeyVaultSettingType type) {
+        this.name = name;
+        this.value = value;
+        this.type = type;
+    }
+
+    /**
+     * Gets the name of the {@link KeyVaultSetting}.
+     *
+     * @return The name of the {@link KeyVaultSetting}.
+     */
+    public String getName() {
+        return this.name;
+    }
+
+    KeyVaultSetting setName(String name) {
+        this.name = name;
+
+        return this;
+    }
+
+    /**
+     * Get the value of the {@link KeyVaultSetting}.
+     *
+     * @return The value of the {@link KeyVaultSetting}.
+     */
+    public String getValue() {
```

Comment:
I don't think we should expose this as a string. See my other comment.

Reply:
If what the service team said holds true and `value` is guaranteed to be a primitive, we could have `getValue()` internally parse the string into the right type indicated by `KeyVaultSettingType` (or `SettingType`).

Reply:
We could also have a convenience method for getting the value as originally received from the service (`String`).

Reply:
I wouldn't do this in .NET. Returning a `bool` (a value type) as an `object` boxes it - allocates memory on the heap. It's also not intuitive to always cast. See my PR for what I'm proposing at least for .NET, if not other languages.